Sources/SourceControl/RepositoryManager.swift

```diff
@@ -384,6 +384,15 @@ public class RepositoryManager {
             } catch {
                 // Fetch without populating the cache in the case of an error.
                 print("Skipping cache due to an error: \(error)")
+                // It is possible that we already created the directory before failing, so clear leftover data if present.
+                do {
+                    try fileSystem.removeFileTree(repositoryPath)
+                } catch let error as NSError {
+                    // If we failed because the directory doesn't actually exist anymore, ignore the error.
+                    if !(error.domain == NSCocoaErrorDomain && error.code == NSFileNoSuchFileError) {
+                        throw error
+                    }
+                }
```

Comment:
so given https://github.com/apple/swift-tools-support-core/pull/186 the `main` version of this will be different, right?

Reply:
Yep, that's right, the `main` version doesn't do the `catch` here and instead relies on the updated `removeFileTree` in TSC to handle the potential race after the `exists` check.